src/services/message.service.ts

```diff
@@ -31,6 +31,48 @@ export class MessageService {
     this._sendConstructedReport(report, { files: message.attachments.map((e) => e.url) });
   }
 
+  async sendTextMessage(
+    message: IMessage,
+    content: string,
+    argOptions: ITextMessageOptions
+  ): Promise<void> {
+    const options: ITextMessageOptions = {};
+    options.header = argOptions.header ? `${argOptions.header}\n` : '';
+    options.footer = argOptions.footer ? `\n${argOptions.footer}` : '';
+    options.reply = argOptions.reply || false;
+    options.delimiter = argOptions.delimiter || '';
+
+    const replyLength = options.reply ? `${message.author},  `.length : 0;
+    const templateLength = options.header.length + options.footer.length;
```

Comment:
Not sure where you got 2048 from, but in testing and what I have seen online 2000 is the solid cap for discord message length.  Also thought about it while I was coding it, but decided not to simply because this route only takes 1 more variable.

Reply:
idk, you are probably right. Nonetheless, If you just give yourself some buffer, it makes things a lot less complex
```ts
const chunks = [];
const split = content.split('\n');

while (split.length > 0) {
  let temp = '';
  while (temp.length < 1000 && split.length > 0) {
    temp = `${split.pop()}\n` + temp;
  }
  chunks.push(temp);
}

await Promise.all(
  chunks.reverse().map((c) => {
    return message.channel.send(`${options.header}${c}${options.footer}`);
  })
);
```
Something along those lines

Reply:
I can do that if you think that is a better method, although I think this way is more like sweeping the problem under the rug.  Since I thought the key idea of this service was to catch when messages are too big, opposed to simply splitting them up which is the primary function of that method.  Although if you think the primary issue with the current code is it's too complex, then I can try to encapsulate it a bit more and change around some variables to try and make it read better.

Reply:
How is this sweeping it under the rug? If the message is too big, it splits it. You can tweak it a little bit, you can make it 1900 if you want, Im just saying if you give yourself reasonable headroom the code becomes a lot simpler.

Reply:
if(not over the limit){
dont nothing
}